code/Items/Inventory.cs

```diff
@@ -163,7 +131,22 @@ public ItemStack GetItem(int position)
                 throw new IndexOutOfRangeException($"There is only {SlotsCount} slots in the inventory.");
             }
 
-            return items[position];
+            return Slots[position].ItemStack;
+        }
+
+        public Slot GetFirstEmptySlot()
+        {
+            return Slots.FirstOrDefault(slot => slot.IsEmpty());
```

Comment:
In the case where no empty slot is found what is returned ?

Reply:
null